docs/core-concepts.md

```diff
@@ -0,0 +1,148 @@
+# Core Concepts
+Mastering Mavericks only requires using three classes: MavericksState, MavericksViewModel, MavericksView.
+
+# MavericksState
+The first step in creating a Mavericks screen is to model it as a function of state. Modeling a screen as a function of state is a useful concept because it is:
+1. Easy to test
+1. Easy for you and other engineer to reason through
+1. Renders the same independently of the order of events leading up to it
+1. Powerful enough to render any type of screen
+
+Mavericks will also enforce that your state class:
+1. Is a kotlin data class
+1. Uses only immutable properties
+1. Has default values for every property to ensure that your screen can be rendered immediately
+
+This concept makes reasoning about and testing a screen trivially easy because given a state class, you can have high confidence that you screen will look correct.
```

Comment:
```suggestion
This concept makes reasoning about and testing a screen trivially easy because given a state class, you can have high confidence that your screen will look correct.
```